elements/pf-checkbox/pf-checkbox.ts

```diff
@@ -0,0 +1,218 @@
+ * @cssprop --pf-c-check__input--MarginTop {@default calc(((var(--pf-c-check__label--FontSize) * var(--pf-c-check__label--LineHeight)) - var(--pf-c-check__input--Height)) / 2)}
+ * @cssprop --pf-c-check__description--FontSize {@default var(--pf-global--FontSize--sm, 0.875rem)}
+ * @cssprop --pf-c-check__description--Color {@default var(--pf-global--Color--200, #6a6e73)}
+ * @cssprop --pf-c-check__body--MarginTop {@default var(--pf-global--spacer--sm, 0.5rem)}
+ * @cssprop --pf-c-check__label-required--MarginLeft {@default var(--pf-global--spacer--xs, 0.25rem)}
+ * @cssprop --pf-c-check__label-required--FontSize {@default var(--pf-global--FontSize--sm, 0.875rem)}
+ * @cssprop --pf-c-check__label-required--Color {@default var(--pf-global--danger-color--100)}
+ */
+@customElement('pf-checkbox')
+export class PfCheckbox extends LitElement {
+  static readonly styles = [styles];
+
+  static formAssociated = true;
+
+  static shadowRootOptions: ShadowRootInit = { ...LitElement.shadowRootOptions, delegatesFocus: true };
+
+  /**
+   * Label text of the checkbox.
+   */
+  @property() label?: string;
+
+  /**
+   * Form value of the checkbox
+   */
+  @property() value = '';
+
+  /**
+   * Flag to show if the checkbox is indeterminate.
+   */
+  @property({ type: Boolean, reflect: true }) indeterminate = false;
+
+  /**
+   * Flag to show if the checkbox is checked.
+   */
+  @property({ type: Boolean, reflect: true }) checked = false;
+
+  /**
+   * Flag to show if the checkbox is disabled.
+   */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /**
+   * Flag to show if the checkbox is required.
+   */
+  @property({ type: Boolean, reflect: true }) required = false;
+
+  @queryAssignedElements() private nestedElements?: HTMLElement[];
+
+  /**
+   * Flag to show if the checkbox selection is valid or invalid.
+   */
+  get valid() {
+    return this.#internals.validity.valid;
+  }
+
+  #internals = this.attachInternals();
+
+  #slots = new SlotController(this, null, 'description', 'body');
+
+  #batching = false;
+
+  constructor() {
+    super();
+    this.addEventListener('restore', e => this.#onRestore(e));
+  }
+
+  override willUpdate() {
+    this.#internals.setFormValue(this.checked ? this.value : null);
+    this.#setDisabledStateOnNestedControls();
+  }
+
+  override render() {
+    const { checked, disabled, indeterminate, required } = this;
+    const emptyNested = !this.#slots.hasSlotted(SlotController.anonymous as unknown as string);
+    const emptyDescription = !this.#slots.hasSlotted('description');
+    const emptyBody = !this.#slots.hasSlotted('body');
+    return html`
+      <input id="checkbox"
+             type="checkbox"
+             aria-labelledby="accessibleLabel"
+             @input="${this.#onChange}"
+             .disabled="${disabled}"
+             .checked="${checked}"
+             .indeterminate="${indeterminate}">
```

Comment:
Needs an `aria-checked` value of `mixed` if indeterminate.

If nested checkboxes, it would need an `aria-controls` with space-separated IDrefs for each nested checkbox. Not sure how we'll do this without cross-root ARIA unless the component's internals is capturing the IDRefs of the slotted checkboxes.

See https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/examples/checkbox-mixed/#rps_label